exp/api/v1beta1/ipaddress_types.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```

Comment:
(moved this comment to a conversation, general PR comments are easily overlooked)

It's a bit unfortunate that we have to start with v1beta1 given that the apiGroup is already at v1beta1 (given that Kubernetes probably doesn't really support sparse apiGroups with CRDs).

I wonder if in general we think this should have it's own apiGroup (e.g. ipam.cluster.x-k8s.io). I wouldn't decide it because of that, but this would allow us to start with v1alpha1 and iterate with less restrictions.

If we put it into the exp group (which is actually just `cluster.x-k8s.io`) bumping the apiVersion for the IPAM types requires bumping the apiVersion of all our other "core" types (Cluster,Machine,...) as well.

Reply:
cc @fabriziopandini @vincepri

Reply:
Below my comments on this topic:

- Due to historical reason, we exp folder contains MP experimental feature, but every other feature we added subsequently are in an exp subfolder. Should this go in exp/ipam in order to be consistent with the recent wave of experimental features? Also, if this goes into a different folder, our package folder will be cleaner and it opens up to have a different API group...
- Machine Pool was originally implemented in a different API group, but then we did a breaking change to bring them into CAPI main API group, which is where we expect them to lend when the experiment moves to GA. given this precedent:
  - I would like to avoid another breaking change for users, so IPAM types should start in the API group we expect them to land at the end of graduation
  - Given that ipam is a well-scoped feature that someone can use or not, I don't see a counter-argument for having this separated from the core API, e.g. in the ipam.cluster.x-k8s.io group starting with v1alpha1 version, but I think this decision should be approved in the office hours.

Reply:
+1 to exp/ipam (side note: our folder structure is a mess, but that's the best we can do at this point)
+1 to `ipam.cluster.x-k8s.io`, `v1alpha1` and bringing it up in the office hours

Reply:
To the first point: long-term it could be an option to switch the APIs on root (not KCP/CABPKG/CAPD) to something which aligns to https://github.com/kubernetes/api (essentially an api/apis folder and then a sub-folder per apiGroup). But this obviously needs a separate issue/discussion

Reply:
The current ipam-provider-in-cluster implementation is also using `ipam.cluster.x-k8s.io`. I guess it's fine to spread those across projects, since `infrastructure.cluster.x-k8s.io` is also shared.

Reply:
> The current ipam-provider-in-cluster implementation is also using ipam.cluster.x-k8s.io. I guess it's fine to spread those across projects, since infrastructure.cluster.x-k8s.io is also shared.

Yup sounds fine to me as well. In case of `infrastructure.cluster.x-k8s.io` a nice effect was that it simplifies RBAC for the core controller when accessing infra resources.

Reply:
Looks like we're fine with moving the code + apiGroup. I'll resolve